File: 백준/Silver/10773. 제로/제로.cc
#include <bits/stdc++.h>
using namespace std;
stack<int> stk;
int n, k;
int main() {
    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> k;
        if(k != 0) {
            stk.push(k);
        }
        else {
            stk.pop();
        }
    }
    int sum = 0;
    while(!stk.empty()) {
        sum += stk.top();
        stk.pop();
    }
    cout << sum << '\n';
}

File: 백준/Silver/10773. 제로/제로.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int k = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();

        while (k-- > 0) {
            int num = Integer.parseInt(br.readLine());
            if (num != 0) {
                stack.push(num);
            } else {
                if (!stack.isEmpty()) {
                    stack.pop();
                }
            }
        }

        int sum = stack.stream().mapToInt(Integer::intValue).sum();
        System.out.println(sum);

    }
}